samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -68,12 +71,20 @@ public IncomingMessageEnvelope(SystemStreamPartition systemStreamPartition, Stri
     this.size = size;
   }
 
-  public void setTimestamp(long timestamp) {
-    this.timestamp = timestamp;
+  public void setEventTime(long eventTime) {
```

Comment:
I'd prefer to add these params to constructors instead of setting via setters.

Reply:
done